### PR TITLE
Fix use of LoRA tokenizers

### PR DESCRIPTION
**Pull request by njhill**
This depends on vLLM PR https://github.com/vllm-project/vllm/pull/6227, which is only held up because I need to find time to add a test to it.

### CODE REVIEW

File: src/vllm_tgis_adapter/grpc/grpc_server.py [hunk 1]
```diff
@@ -706,39 +718,36 @@ def _convert_tokens(  # noqa: PLR0913
...
-        input_ids = await self.tokenizer_group.encode_async(
-            prompt,
-            **tokenize_kwargs,
-        )
 
-        # TODO this is temporary until truncation option is added
-        # to the TokenizerGroup encode methods
-        if truncate_input_tokens and truncate_input_tokens < len(input_ids):
-            input_ids = input_ids[-truncate_input_tokens:]
-            if not sampling_params.skip_special_tokens:
```
Comment (njhill) [inline src/vllm_tgis_adapter/grpc/grpc_server.py:734]: This was actually a bug, it shouldn't have been checking this param here (it applies to output tokens not input tokens).

**Comment by dtrifiro** [comment]
Merge after #47

File: src/vllm_tgis_adapter/__main__.py [hunk 2]
```diff
@@ -45,18 +50,30 @@
...
+
+try:
+    from vllm.entrypoints.openai.serving_tokenization import (
+        OpenAIServingTokenization,  # noqa: TCH002
+    )
+except ImportError:  #  vllm<=0.5.2
+    has_tokenization = False
+else:
+    has_tokenization = True
+
```
Comment (dtrifiro) [inline src/vllm_tgis_adapter/__main__.py:62]: This avoids lora breaking for <=0.5.2

File: tests/conftest.py [hunk 3]
```diff
@@ -36,23 +37,44 @@ def monkeysession():
...
+def lora_adapter_name():
+    return "lora-test"
+
+
+@pytest.fixture()
+def lora_adapter_path():
+    from huggingface_hub import snapshot_download
+
+    path = snapshot_download(repo_id="yard1/llama-2-7b-sql-lora-test")
+    return path
```
Comment (dtrifiro) [inline tests/conftest.py:50]: This is the same adapter used in the docs: https://docs.vllm.ai/en/latest/models/lora.html

File: tests/test_grpc_server.py [hunk 4]
```diff
@@ -48,3 +49,10 @@ def test_batched_generation_request(grpc_client, grpc_server_thread_port):
 
     assert len(responses) == 2
     assert all(response.text for response in responses)
+
+
+@pytest.mark.skipif(vllm.config.is_cpu(), reason="Lora is not supported on cpu")
+def test_lora_request(grpc_client, lora_adapter_name):
+    response = grpc_client.make_request("hello", adapter_id=lora_adapter_name)
+
+    assert response.text
```
Comment (dtrifiro) [inline tests/test_grpc_server.py:58]: @njhill just a simple sanity check for the grpc API, is there anything else you would like to test?

**Review by prashantgupta24 — APPROVED** [review]

**Pull request merged**